### ISSUE TITLE
TS error "An import path cannot end with a '.ts' extension."

### ISSUE THREAD

**Issue by Varkal** [minio/minio-js#1173]
Hi ! 

Since I've upgraded to the latest version (7.1.1), my typescript code don't compile anymore with these error messages 

```
node_modules/minio/dist/main/minio.d.ts:13:8 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './helpers' instead.

13 } from './helpers.ts'
          ~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:14:36 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/client' instead.

14 import type { ClientOptions } from './internal/client.ts'
                                      ~~~~~~~~~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:15:32 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/copy-conditions' instead.

15 import { CopyConditions } from './internal/copy-conditions.ts'
                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:16:28 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/post-policy' instead.

16 import { PostPolicy } from './internal/post-policy.ts'
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:17:29 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/s3-endpoints' instead.

17 import type { Region } from './internal/s3-endpoints.ts'
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:19:15 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './helpers' instead.

19 export * from './helpers.ts'
                 ~~~~~~~~~~~~~~

node_modules/minio/dist/main/minio.d.ts:20:29 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/s3-endpoints' instead.

20 export type { Region } from './internal/s3-endpoints.ts'
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

`allowImportingTsExtensions` in tsconfig don't fix the issue as my project is in typescript and i cannot enable `noEmit` or `emitDeclarationOnly`. 

Currently the only way I've found to continue using minio is to rollback the upgrade and pin the version

**Comment by Cocophotos** [comment]
Same issue for me

**Comment by prakashsvmx** [comment]
Please share details about the project setup.  (node version, tsc version etc), you could wait for the next release as well as typescript migration is in progress.

**Comment by Cocophotos** [comment]
> Please share details about the project setup. (node version, tsc version etc), you could wait for the next release as well as typescript migration is in progress.

I've updated my comment! Thx

**Comment by prakashsvmx** [comment]
The fix is in https://github.com/minio/minio-js/pull/1167
Please wait for a newer release.

**Comment by Varkal** [comment]
Wow nice ! Thanks you so much ! Have you any estimation about the release date ?

**Comment by Nicolas-CARRERE** [comment]
Hi all, i m certainly too young in my short web dev experience, but i don't know how to set up the mentionned in the previous @prakashsvmx  post 
The fix is in #1167 🧐
Can you please explain a little bit more ? 🙏

**Comment by trim21** [comment]
> Hi all, i m certainly too young in my short web dev experience, but i don't know how to set up the fix mentionned in the previous @prakashsvmx post The fix is in #1167 🧐 Can you please explain a little bit more ? 🙏

tldr: wait a new release

**Comment by Christiaan-Mathu** [comment]
Hi @prakashsvmx do you have an ETA for this release or maybe a work around for the meanwhile?

Thanks in advance :)

**Comment by harishoss** [comment]
Not sure if this will work for everyone, Try adding `"skipLibCheck": true` in tsconfig.json

**Comment by trim21** [comment]
try latest version

**Comment by prakashsvmx** [comment]
Fix part of https://github.com/minio/minio-js/releases/tag/7.1.2
